File: packages/gyroscope/android/src/main/java/io/capawesome/capacitorjs/plugins/gyroscope/classes/results/IsAvailableResult.java
package io.capawesome.capacitorjs.plugins.gyroscope.classes.results;

import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.gyroscope.classes.interfaces.Result;

public class IsAvailableResult implements Result {

    private final boolean available;

    public IsAvailableResult(boolean available) {
        this.available = available;
    }

    @Override
    public JSObject toJSObject() {
        JSObject result = new JSObject();
        result.put("available", available);
        return result;
    }
}

File: packages/age-signals/ios/Plugin/Classes/Results/IsAvailableResult.swift
import Capacitor

@objc class IsAvailableResult: NSObject, Result {
    private let available: Bool

    init(available: Bool) {
        self.available = available
    }

    func toJSObject() -> AnyObject {
        var result = JSObject()
        result["available"] = available
        return result as AnyObject
    }
}
